lib/internal/util.js

```diff
@@ -153,6 +153,15 @@ function slowCases(enc) {
   }
 }
 
+function isValidEncoding(val, encoding) {
```

Comment:
Since this is a validator now...it would make more sense to move this into `lib/internal/validators` and just throw the error from there. Since we are going semver-major anyway, we could just provide more information when we construct the error (e.g. instead of saying `'encoding' is invalid for data`, say `'encoding' is invalid for data of length ${length}` - that may still be somewhat cryptic for users, but better than what it currently is)

Reply:
Fixed.